bot.py

```diff
@@ -52,10 +58,88 @@
 tree = app_commands.CommandTree(client)
 db = ScheduleDB(SCHEDULE_DB_PATH)
 pairings_db = PairingsDB(PAIRINGS_DB_PATH)
+leetcode_db = LeetCodeDB(LEETCODE_DB_PATH)
 
 
 # Discord API currently doesn't support variadic arguments
 # https://github.com/discord/discord-api-docs/discussions/3286
+@tree.command(
+    name="leetcode", description="Get a random LeetCode problem based on difficulty."
+)
+@app_commands.describe(difficulty="Choose a difficulty level for the LeetCode problem.")
+@app_commands.choices(
+    difficulty=[
+        app_commands.Choice(name="Easy", value="Easy"),
+        app_commands.Choice(name="Medium", value="Medium"),
+        app_commands.Choice(name="Hard", value="Hard"),
+        app_commands.Choice(name="Any difficulty", value="Any"),
+    ]
+)
+async def _leetcode(interaction: discord.Interaction, difficulty: str):
+    try:
+        problem = get_random_leetcode_problem(difficulty)
+        message = f"Here's a {difficulty} LeetCode problem: {problem['title']} - {problem['url']}"
+        await interaction.response.send_message(message, ephemeral=True)
+    except Exception as e:
+        logger.error(f"Error fetching LeetCode problem: {e}", exc_info=True)
+        await interaction.response.send_message(
+            "Sorry, there was an error fetching the problem.", ephemeral=True
+        )
+
+
+@tree.command(
+    name="leetcode-subscribe",
```

Comment:
We should probably rename the APIs `/subscribe` -> `/subscribe-pairing` and `/leetcode-subscribe` -> `/subscribe-leetcode` for consistency. Whatever name you use for "pairing" should be the same as the flag used in the `/unsubscribe` command (you currently have it as "Programming").

API changes are annoying for users so better to get it right early.